velox/common/base/Status.h

```diff
@@ -476,11 +477,33 @@ void Status::moveFrom(Status& s) {
     VELOX_RETURN_IF(!__s.ok(), __s);                          \
   } while (false)
 
-namespace internal {
+/// Return with given status wrapped in folly::Unexpected if condition is met.
+#define VELOX_RETURN_UNEXPECTED_IF(condition, status) \
+  do {                                                \
+    if (FOLLY_UNLIKELY(condition)) {                  \
+      return (::folly::makeUnexpected(status));       \
```

Comment:
@marin-ma There is a consensus on reducing dependency on Folly. I see that we need something like Arrow's `Result` https://github.com/apache/arrow/blob/main/cpp/src/arrow/result.h
@pedroerp any thought on if we should use folly or maybe implement `Result` in Velox?

Reply:
@majetideepak we already have quite a few usages of folly::Expected throughout the library. My suggestion would be to use that (at least for now) for consistency, and if we ever want to migrate off of it we could do it in a separate PR.